clients/src/main/java/org/apache/kafka/common/requests/MetadataRequest.java

```diff
@@ -25,24 +25,42 @@
 import java.util.List;
 
 public class MetadataRequest extends AbstractRequest {
-    
+
     private static final Schema CURRENT_SCHEMA = ProtoUtils.currentRequestSchema(ApiKeys.METADATA.id);
     private static final String TOPICS_KEY_NAME = "topics";
 
+    private final boolean allTopics;
     private final List<String> topics;
 
     public MetadataRequest(List<String> topics) {
+        this(topics, false);
+    }
+
+    /**
+     * If allTopics is true, then the passed topics will be ignored.
+     */
+    public MetadataRequest(List<String> topics, boolean allTopics) {
```

Comment:
In current proposal, we are saying a null indicates all topics. Would it be possible to use that here as well?